server/src/main/java/io/littlehorse/common/model/getable/core/noderun/NodeRunModel.java

```diff
@@ -328,29 +328,29 @@ public NodeRun.Builder toProto() {
     }
 
     public boolean isInProgress() {
-        return (status != LHStatus.COMPLETED && status != LHStatus.HALTED && status != LHStatus.ERROR);
+        switch (status) {
+            case STARTING:
+            case RUNNING:
+            case HALTING:
+                return true;
+            case HALTED:
+            case COMPLETED:
+            case EXCEPTION:
+            case ERROR:
+            case UNRECOGNIZED:
+        }
+        return false;
     }
 
     public boolean isCompletedOrRecoveredFromFailure() {
         if (status == LHStatus.COMPLETED) {
             return true;
         }
 
-        if (status == LHStatus.ERROR || status == LHStatus.HALTED || status == LHStatus.EXCEPTION) {
```

Comment:
This super-complex logic is no longer needed now that we fixed the `EXCEPTION` issue above.